gluten-core/src/main/scala/io/glutenproject/execution/HashAggregateExecBaseTransformer.scala

```diff
@@ -73,6 +74,24 @@ abstract class HashAggregateExecBaseTransformer(
       aggregateAttributes)
   }
 
+  protected def isGroupingKeysPreGrouped: Boolean = {
+    if (!conf.getConf(GlutenConfig.COLUMNAR_PREFER_STREAMING_AGGREGATE)) {
+      return false
+    }
```

Comment:
The check doesn't do as the method name `isGroupingKeysPreGrouped` said?

I would suggest renaming `isGroupingKeysPreGrouped` to something like `isCapableForStreamingAggregation`, or just moving this config check out.